From: lichenggang <lichenggang@deepin.org>
Date: Tue, 06 May 2026 10:00:00 +0800
Subject: [PATCH] SConstruct: Add support for sw64 architecture

Add __sw_64__ to the little-endian architecture detection list in
AppendEndianCheck() so that the build does not fail with:
  #error can not detect the endianness!

Forwarded: not-needed

---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index e97bba98..f1a2b3c4 100644
--- a/SConstruct
+++ b/SConstruct
@@ -349,7 +349,7 @@ def AppendEndianCheck(conf):
   || defined(_M_X64)    || defined(__bfin__) \
   || defined(__alpha__) || defined(__ARMEL__) \
   || defined(_MIPSEL)   || (defined(__sh__) && defined(__LITTLE_ENDIAN__)) \
-  || defined(__riscv)   || defined(__loongarch__) \
+  || defined(__riscv)   || defined(__loongarch__)   || defined(__sw_64__) \
   || defined(__AARCH64EL__)
 # undef WORDS_BIGENDIAN

